pandas/io/clipboard.py

```diff
@@ -86,6 +89,12 @@ def to_clipboard(obj, excel=None, sep=None, **kwargs):  # pragma: no cover
             obj.to_csv(buf, sep=sep, **kwargs)
             clipboard_set(buf.getvalue())
             return
+        except UnicodeEncodeError:
+            # try again with encoding from locale
+            from locale import getdefaultlocale
+            obj.to_csv(buf, sep=sep, encoding=getdefaultlocale()[1], **kwargs)
```

Comment:
It is a while ago since I looked at this. I just remember that I found some edge cases which still failed. I'll try to find my test cases to see if what you propose works.